### DISCUSSION TITLE
Saved workflows with HttpEndpoint trigger can't be executed with http path after project startup

### DISCUSSION THREAD

**Discussion by abhjitsantra** [elsa-workflows/elsa-core#4354] [Q&A] [answered]
We are using ELSA v3. When we try to execute a workflow (that starts with an HttpEndpoint activity) by invoking the path the server returns 404 status code. This is happening for the already created workflows those are saved in the SQL database. Each time after project startup we are to publish the workflows again so that they can be executed by the path exposed by the HttpEndpoint activity.

This was not the case with 3.0.0-preview-301. After upgrading to 3.0.0-preview-659 we are facing this problem. What could be the reason behind it?

**Comment by sfmskywalker** [comment]
Hello @abhjitsantra,

Thank you for reporting the issue.

There have been significant changes recently that might have broken something.

If you don't mind, please share with me steps to reproduce. This could be in the form of a sample workflow JSON file, and or a sample project. This will help us better understand and address the issue.

**Reply by abhjitsantra** [reply]
Hello @sfmskywalker,

Please find attached sample project with the problem I mentioned. I have created a workflow from the designer and published it. The workflow has http endpoint trigger path `wfl-test`, so I can execute the workflow from a rest client by hitting the URL `https://localhost:5001/workflows/wfl-test`. As soon as I restart the application, I can't execute the workflow again by hitting the URL until I publish it again. Please note that, the workflow can be executed using the workflow definition Id and need not be published each time after application restart.

I am attaching the workflow JSON file and the sample project for your reference.

Workflow Json: [workflow-test.zip](https://github.com/elsa-workflows/elsa-core/files/12452551/workflow-test.zip)
Solution: [ElsaWorkflow.zip](https://github.com/elsa-workflows/elsa-core/files/12452560/ElsaWorkflow.zip)

**Reply by Eatric** [reply]
Yep, we have the same problem with PostgreSQL and RC1

**Reply by sfmskywalker** [reply, marked as answer]
Thanks @abhjitsantra for the sample project! I was able to reproduce the issue, and I'm happy to report that it has been solved with the latest version. After updating to the latest version, you'll find that there are some breaking changes in terms of code. For your convenience, please find a modified version of Program.cs that should work:

```csharp
using Elsa.EntityFrameworkCore.Extensions;
using Elsa.EntityFrameworkCore.Modules.Labels;
using Elsa.EntityFrameworkCore.Modules.Management;
using Elsa.EntityFrameworkCore.Modules.Runtime;
using Elsa.Extensions;
using Microsoft.AspNetCore.Mvc;

namespace ElsaWorkflow
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var builder = WebApplication.CreateBuilder(args);

            // Add services to the container.
            builder.Services.AddElsa(elsa =>
            {
                var elsaServerConn = builder.Configuration.GetConnectionString("ElsaServer") ?? string.Empty;

                elsa
                .UseWorkflowManagement(conf =>
                {
                    conf.UseEntityFrameworkCore(ef => ef.UseSqlServer(elsaServerConn));
                })
                .UseWorkflowRuntime(conf =>
                {
                    conf.UseEntityFrameworkCore(ef => ef.UseSqlServer(elsaServerConn));
                    conf.UseDefaultRuntime();
                })
                .UseLabels(conf => conf.UseEntityFrameworkCore(ef => ef.UseSqlServer(elsaServerConn)))
                .UseWorkflowsApi()
                .UseLiquid()
                .UseJavaScript()
                .UseHttp(http => http.ConfigureHttpOptions = options => options.BasePath = "/workflows")
                .UseIdentity(identity =>
                {
                    identity.UseAdminUserProvider();
                    identity.TokenOptions = options => options.SigningKey = "secret-token-signing-key";
                })
                .UseDefaultAuthentication(auth => auth.UseAdminApiKey());
            });

            builder.Services.AddCors(opt =>
            {
                opt.AddDefaultPolicy(policy =>
                {
                    policy.AllowAnyHeader().AllowAnyMethod().AllowAnyOrigin();
                });
            });
            builder.Services.AddRazorPages(options => options.Conventions.ConfigureFilter(new IgnoreAntiforgeryTokenAttribute()));

            var app = builder.Build();

            app.UseHttpsRedirection();
            app.UseStaticFiles();
            app.UseAuthentication();
            app.UseAuthorization();
            app.UseWorkflowsApi();
            app.UseWorkflows();
            app.MapRazorPages();

            app.Run();
        }
    }
}
```

If you run into any issues, please let me know.

Thanks for your contributions!

@Eatric Thank you for confirming the issue - should you run into any issues after upgrading, please do let me know.

**HEADS UP** the latest version introduces schema changes - please make sure to backup your data and export your workflows before  resetting your databases.
